src/main/java/uk/gov/companieshouse/web/accounts/interceptor/CompanyAccountsDataStateInterceptor.java

```diff
@@ -0,0 +1,180 @@
+package uk.gov.companieshouse.web.accounts.interceptor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.security.SignatureException;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.companieshouse.web.accounts.model.state.AccountingPolicies;
+import uk.gov.companieshouse.web.accounts.model.state.CompanyAccountsDataState;
+import uk.gov.companieshouse.web.accounts.model.state.CompanyAccountsDataStates;
+import uk.gov.companieshouse.web.accounts.token.TokenManager;
+
+@Component
+public class CompanyAccountsDataStateInterceptor extends HandlerInterceptorAdapter {
+
+    private static final String STATE_COOKIE_NAME = "__CAS";
```

Comment:
Is there a reason for the double underscore?

Reply:
Yes, it complies with the standard naming convention for Companies House cookies

Reply:
👍